03_Fiches_thematiques/Fiche_api.Rmd

```diff
@@ -0,0 +1,528 @@
+diffusées sur <https://www.insee.fr> dans la rubrique *Chiffres détaillés*,
+sous forme de cubes prédéfinis. Cette API est hébergée sur le catalogue des
+API de l'Insee. Une authentification par un jeton est donc nécessaire.
+
+Le _package_ comporte une fonction unique qui permet d’importer les données
+présentes dans l’API *Données Locales* dans une liste contenant 4 objets :
+
+* les données statistiques ;
+* les modalités de chaque variable ;
+* l’information sur la zone demandée ;
+* l’information sur la source et le jeu de données demandé.
+
+
+Exemple d'utilisation du _package_ pour importer le nombre
+d'entreprises et d'établissements en 2017 (en géographie au 01/01/2017)
+selon l'activité en 5 catégories et une indicatrice indiquant s'il s'agit
+d'une entreprise individuelle ou non pour la commune de Nantes:
+
+```{r, eval = FALSE}
+library(inseeLocalData)
+
+croisement <- "NA5_B-ENTR_INDIVIDUELLE"
+jeu_donnees <- "GEO2017REE2017"
+nivgeo <- "COM"
+codgeo <- "44109" #CODE GEO DE NANTES
+modalite <- "all.all"
+
+donneesAPI <- get_dataset(jeton, jeu_donnees, croisement, modalite, nivgeo, codgeo)
+
+donnees <- donneesAPI$donnees # pour accéder aux données
+liste_code <- donneesAPI$liste_code # pour accéder aux nomenclatures
+info_zone <- donneesAPI$info_zone # pour accéder aux données géographiques
+source <- donneesAPI$source # pour accéder à la source
+
+```
+
+*TO BE COMPLETED: il faudrait plus de détails*
+
+
+### [insee](https://github.com/InseeFr/R-Insee-Data/)
+
+Ce _package_ permet de télécharger les données et leurs métadonnées
+diffusées sur le service `SDMX` de la Base de données Macroéconomique
+de l'Insee (BDM). Cette API étant ouverte, son accès ne demande pas
+d'identification, ni de jeton. Il est uniquement nécessaire de déterminer
+les données souhaitées soit via une liste de catégories (`idbank`),
+soit via une liste de données (`dataset`). 
+
+```{r echo = FALSE, message = FALSE, warning = FALSE}
+library(magrittr)
+df <- data.frame("Information" = c("Liste des jeux de données",
+                                   "Liste des séries"),
+                 "Fonction" = c("`insee::get_dataset_list()`",
+                                "`insee::get_idbank_list()`")
+)
+df %>% 
+  knitr::kable(escape = F, position = "center", full_width = F, align="ccl") %>%
+  kableExtra::column_spec(1, width = "3cm", bold = TRUE) %>%
+  kableExtra::column_spec(2, width = "2cm")
+
+```
+
+
+
+
+```{r, eval = FALSE}
+library(insee)
+
+# Importer les données à partir de leur idbank ou importer un dataset à partir de leur identifiant:
+table_bp <- get_insee_dataset("BALANCE-PAIEMENTS")
+indicateur_001694056 <- get_insee_idbank("001694056")
+```
+
+Il est possible de rajouter des filtres à ces fonctions afin de limiter
+le nombre de données importées (filtre sur la période, sur la date de mise à jour, sur les filtres).
+
+
+**TO BE COMPLETED**
+
+
+### [doremifasol](https://github.com/InseeFrLab/DoReMIFaSol)
```

Comment:
Je regrouperais `doremifasol` avec `apinsee` puisque ce dernier est une dépendance de ce premier ; par ailleurs, si je comprends bien, le seul package parmi ceux listé à utiliser `apinsee`.

Reply:
j'ai déplacé la section `doremifasol` avec 60dce25cc8edef48a9a568d3faa352531dd6273f

Reply:
quant à la recommandation, j'ai essayé d'être agnostique par rapport aux différents packages qui existent. Je suis pas fermé à en recommander un plutôt qu'un autre, si tu me dis que `doremifasol` est un peu le "one ring to rule them all"

Reply:
Non je n'irai pas sur ce terrain-là. Chaque package a sa place et son usage, à mon avis.

Reply:
> Je regrouperais `doremifasol` avec `apinsee` puisque ce dernier est une dépendance de ce premier ; par ailleurs, si je comprends bien, le seul package parmi ceux listé à utiliser `apinsee`.

Actuellement oui, mais ne peut-on pas imaginer que `apinsee` soit utilisé à terme par `inseeLocalData` et `insee` (vu que c'est tellement pratique, ça me semble plausible) ? Ton avis @RLesur ?

Dans ce cas de figure hypothétique, il vaudrait mieux le laisser séparé de `doremifasol`.